src/length-value.js

```diff
@@ -65,11 +64,44 @@
   };
 
   LengthValue.prototype.fromValue = function(value, type) {
-    throw new TypeError('Not implemented yet');
+    return new SimpleLength(value, type);
   };
 
+  /**
+   * Creates a LengthValue from a CalcDictionary.
+   * Note if the dictionary contains:
+   *  - 0 valid lengths, an error is thrown
+   *  - 1 valid length, a SimpleLength object is created
+   *  - >1 valid lengths, a CalcLength object is created
+   */
   LengthValue.prototype.fromDictionary = function(dictionary) {
-    throw new TypeError('Not implemented yet');
+    if (typeof dictionary != 'object')
+      throw new TypeError('fromDictionary must be passed a CalcDictionary object.');
+
+    var firstValidType = null;
+    var numLengthsFound = 0;
+    for (var type in LengthValue.LengthType) {
+      var value = dictionary[type];
+      if (typeof value == 'number') {
+        numLengthsFound++;
+        if (numLengthsFound > 1) {
+          break; // Found more than one valid length.
+        }
+        // Otherwise, it is the first valid length found.
+        firstValidType = type;
+      } else if (value != null && value != undefined) {
+        throw new TypeError('Value of each length must be null or a number.');
+      }
+    }
+
+    if (numLengthsFound < 1) {
+      throw new TypeError('A CalcDictionary must have at least one valid length.');
+    } else if (numLengthsFound < 2) {
+      return new SimpleLength(dictionary[firstValidType], firstValidType);
+    } else {
+      return new CalcLength(dictionary);
```

Comment:
As discussed, we don't want to do this any more - you can move all of this logic into the constructor, and get rid of the part where you create a SimpleLength.
Sorry about that...